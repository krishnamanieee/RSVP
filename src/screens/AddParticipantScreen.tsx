import React from 'react';
import {ScrollView, View, Button, Text, StyleSheet} from 'react-native';
import DatePicker from 'react-native-datepicker';
import TextInput from '../components/TextInput';
import Theme from '../theme';
import LoadingSpinner from '../components/LoadingSpinner';
import {LocalStorage, LocalStorageKeys} from '../utils/LocalStorage';

interface IErrorState {
  name: string;
  age: string;
  date: string;
  locality: string;
  numberOfGuest: string;
  address: string;
}

interface IAddParticipantState {
  name: string;
  age: string;
  date: string;
  locality: string;
  numberOfGuest: string;
  address: string;
  loading: boolean;
  errors: IErrorState;
}

interface IVailFormReturn {
  status: boolean;
  errors: IErrorState;
}

const initialError = {
  name: '',
  age: '',
  date: '',
  locality: '',
  numberOfGuest: '',
  address: '',
};

class AddParticipantScreen extends React.PureComponent<
  any,
  IAddParticipantState
> {
  public state = {
    name: '',
    age: '',
    date: '',
    locality: '',
    numberOfGuest: '',
    address: '',
    loading: false,
    errors: initialError,
  };

  render(): React.ReactElement {
    const {
      name,
      address,
      age,
      errors,
      locality,
      numberOfGuest,
      date,
      loading,
    } = this.state;
    const onNameChange = (value: string) => this.onChangeValue('name', value);
    const onAgeChange = (value: string) => this.onChangeValue('age', value);
    const onDateChange = (value: string) => this.onChangeValue('date', value);
    const onLocalityChange = (value: string) =>
      this.onChangeValue('locality', value);
    const onNumberOfGuestChange = (value: string) =>
      this.onChangeValue('numberOfGuest', value);
    const onAddressChange = (value: string) =>
      this.onChangeValue('address', value, 50);
    return (
      <View style={styles.container}>
        <ScrollView contentContainerStyle={{padding: 5, marginVertical: 20}}>
          <TextInput
            title={'Name'}
            value={name}
            onChangeText={onNameChange}
            placeholder={'Please Enter the Name'}
            errorMessage={errors.name}
          />
          <TextInput
            title={'Age'}
            value={age}
            onChangeText={onAgeChange}
            placeholder={'Please Enter the age'}
            errorMessage={errors.age}
            maxLength={2}
            keyboardType={'number-pad'}
          />
          <View style={{margin: 5}}>
            <Text style={{fontSize: Theme.font.text.text, paddingVertical: 5}}>
              {'Date'}
            </Text>
            <DatePicker
              style={{width: 200}}
              date={date}
              mode="date"
              placeholder="select date"
              format="YYYY-MM-DD"
              minDate="2016-05-01"
              maxDate="2016-06-01"
              confirmBtnText="Confirm"
              showIcon={false}
              cancelBtnText="Cancel"
              customStyles={{
                dateInput: {
                  backgroundColor: Theme.colors.input.background,
                  borderWidth: 0,
                },
                placeholderText: {
                  color: Theme.colors.mediumGray,
                },
              }}
              onDateChange={onDateChange}
            />
          </View>
          <TextInput
            title={'Locality'}
            value={locality}
            onChangeText={onLocalityChange}
            placeholder={'Please Enter the Locality'}
            errorMessage={errors.locality}
          />
          <TextInput
            title={'Number of Guests (max 2 members)'}
            value={numberOfGuest}
            onChangeText={onNumberOfGuestChange}
            placeholder={'Please Enter the Number of Guests'}
            errorMessage={errors.numberOfGuest}
            maxLength={1}
            keyboardType={'number-pad'}
          />
          <TextInput
            title={`Address (max 50 ${
              address.length > 0 ? ' : ' + address.length : ''
            } )`}
            value={address}
            onChangeText={onAddressChange}
            placeholder={'Please Enter the Address'}
            errorMessage={errors.address}
            multiline={true}
            style={{height: 50}}
          />
          <View>
            <Button title={'Add'} onPress={this.onSubmitPress} />
          </View>
        </ScrollView>
        <LoadingSpinner visible={loading} />
      </View>
    );
  }

  onChangeValue = (key: string, value: string, length?: number): void => {
    const {errors} = this.state;
    if (length && value.length > length) {
      return;
    }

    // @ts-ignore
    this.setState({
      [key]: value,
      errors: {
        ...errors,
        [key]: '',
      },
    });
  };

  onSubmitPress = async () => {
    const {address, date, numberOfGuest, locality, age, name} = this.state;
    const validFromError: IVailFormReturn = this.isValidData();
    if (validFromError.status) {
      this.setState({errors: {...initialError, ...validFromError.errors}});
      return;
    }
    this.updateLoading();

    setTimeout(async () => {
      const item = {
        id: Math.random().toFixed(0),
        name,
        age,
        DOB: date,
        locality,
        guest: numberOfGuest,
        address,
      };
      let data: any[] | null = [];
      const fetchData = await LocalStorage.get<string>(
        LocalStorageKeys.USER_DATA,
      );
      if (fetchData) {
        data.push(item);
        data.push(...JSON.parse(fetchData));
      } else {
        data.push(item);
      }
      await LocalStorage.set(LocalStorageKeys.USER_DATA, JSON.stringify(data));
      this.updateLoading();
      this.resetTheFrom();
    }, 2000);
    //
  };

  resetTheFrom = () => {
    this.setState({
      name: '',
      age: '',
      date: '',
      locality: '',
      numberOfGuest: '',
      address: '',
    });
  };

  updateLoading = () => {
    const {loading} = this.state;
    this.setState({loading: !loading});
  };

  isValidData = (): IVailFormReturn => {
    const {address, date, numberOfGuest, locality, age, name} = this.state;
    const errors: IErrorState = {...initialError};
    this.setState({errors: {...errors}});
    console.log('[[[98989>>>>>>>>', errors);
    let status: boolean = false;
    if (name.length < 1) {
      status = true;
      errors.name = 'fill the input';
    }
    if (age.length < 1) {
      status = true;
      errors.age = 'fill the input';
    }
    if (date.length < 1) {
      status = true;
      errors.date = 'fill the input';
    }
    if (numberOfGuest.length < 1) {
      status = true;
      errors.numberOfGuest = 'fill the input';
    }
    if (address.length < 1) {
      status = true;
      errors.address = 'fill the input';
    }
    if (locality.length < 1) {
      status = true;
      errors.locality = 'fill the input';
    }
    console.log('err', errors);
    return {
      errors,
      status,
    };
  };
}

export default AddParticipantScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
