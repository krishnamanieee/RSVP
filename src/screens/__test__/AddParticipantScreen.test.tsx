import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import AddParticipantScreen from '../AddParticipantScreen';

const mockFunction = jest.fn();
let props = {};

describe('AddParticipantScreen', () => {
  let wrapper: any;
  let instance: AddParticipantScreen;
  beforeEach(() => {
    wrapper = shallow<AddParticipantScreen, any, any>(
      <AddParticipantScreen {...props} />,
    );
    instance = wrapper.instance();
  });

  it('should match match snapshot', () => {
    const tree = renderer.create(<AddParticipantScreen {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
