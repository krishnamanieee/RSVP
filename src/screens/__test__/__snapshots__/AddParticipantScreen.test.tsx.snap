// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddParticipantScreen should match match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "marginVertical": 20,
        "padding": 5,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Name
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Please Enter the Name"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Age
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="number-pad"
          maxLength={2}
          onChangeText={[Function]}
          placeholder="Please Enter the age"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "margin": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Date
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "width": 142,
              },
              Object {
                "width": 200,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#aaa",
                    "borderWidth": 1,
                    "flex": 1,
                    "height": 40,
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#dadada",
                    "borderWidth": 0,
                  },
                  false,
                  false,
                ]
              }
            >
              <Text
                allowFontScaling={true}
                style={
                  Array [
                    Object {
                      "color": "#c9c9c9",
                    },
                    Object {
                      "color": "#C7C7CD",
                    },
                  ]
                }
              >
                select date
              </Text>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              supportedOrientations={
                Array [
                  "portrait",
                  "portrait-upside-down",
                  "landscape",
                  "landscape-left",
                  "landscape-right",
                ]
              }
              transparent={true}
              visible={false}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "backgroundColor": "#00000077",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        pointerEvents="auto"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "borderTopColor": "#ccc",
                                "borderTopWidth": 1,
                                "marginTop": 42,
                              },
                              undefined,
                            ]
                          }
                        >
                          <RCTDatePicker
                            date={1464719400000}
                            maximumDate={1464719400000}
                            minimumDate={1462041000000}
                            mode="date"
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "height": 216,
                              }
                            }
                            timeZoneOffsetInMinutes={null}
                          />
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "height": 42,
                              "justifyContent": "center",
                              "paddingHorizontal": 20,
                              "position": "absolute",
                              "top": 0,
                            },
                            Object {
                              "left": 0,
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={true}
                          style={
                            Array [
                              Object {
                                "color": "#46cf98",
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#666",
                              },
                              undefined,
                            ]
                          }
                        >
                          Cancel
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "height": 42,
                              "justifyContent": "center",
                              "paddingHorizontal": 20,
                              "position": "absolute",
                              "top": 0,
                            },
                            Object {
                              "right": 0,
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={true}
                          style={
                            Array [
                              Object {
                                "color": "#46cf98",
                                "fontSize": 16,
                              },
                              undefined,
                            ]
                          }
                        >
                          Confirm
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Locality
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Please Enter the Locality"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Number of Guests (max 2 members)
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="number-pad"
          maxLength={1}
          onChangeText={[Function]}
          placeholder="Please Enter the Number of Guests"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Address (max 50  )
        </Text>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Please Enter the Address"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              Object {
                "height": 50,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Add
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`HomeScreen should match match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "marginVertical": 20,
        "padding": 5,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Name
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Please Enter the Name"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Age
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="number-pad"
          maxLength={2}
          onChangeText={[Function]}
          placeholder="Please Enter the age"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "margin": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Date
        </Text>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "width": 142,
              },
              Object {
                "width": 200,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderColor": "#aaa",
                    "borderWidth": 1,
                    "flex": 1,
                    "height": 40,
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#dadada",
                    "borderWidth": 0,
                  },
                  false,
                  false,
                ]
              }
            >
              <Text
                allowFontScaling={true}
                style={
                  Array [
                    Object {
                      "color": "#c9c9c9",
                    },
                    Object {
                      "color": "#C7C7CD",
                    },
                  ]
                }
              >
                select date
              </Text>
            </View>
            <Modal
              animationType="none"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              supportedOrientations={
                Array [
                  "portrait",
                  "portrait-upside-down",
                  "landscape",
                  "landscape-left",
                  "landscape-right",
                ]
              }
              transparent={true}
              visible={false}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "backgroundColor": "#00000077",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={false}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fff",
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        pointerEvents="auto"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "borderTopColor": "#ccc",
                                "borderTopWidth": 1,
                                "marginTop": 42,
                              },
                              undefined,
                            ]
                          }
                        >
                          <RCTDatePicker
                            date={1464719400000}
                            maximumDate={1464719400000}
                            minimumDate={1462041000000}
                            mode="date"
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "height": 216,
                              }
                            }
                            timeZoneOffsetInMinutes={null}
                          />
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "height": 42,
                              "justifyContent": "center",
                              "paddingHorizontal": 20,
                              "position": "absolute",
                              "top": 0,
                            },
                            Object {
                              "left": 0,
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={true}
                          style={
                            Array [
                              Object {
                                "color": "#46cf98",
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#666",
                              },
                              undefined,
                            ]
                          }
                        >
                          Cancel
                        </Text>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "height": 42,
                              "justifyContent": "center",
                              "paddingHorizontal": 20,
                              "position": "absolute",
                              "top": 0,
                            },
                            Object {
                              "right": 0,
                            },
                            undefined,
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={true}
                          style={
                            Array [
                              Object {
                                "color": "#46cf98",
                                "fontSize": 16,
                              },
                              undefined,
                            ]
                          }
                        >
                          Confirm
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Locality
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Please Enter the Locality"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Number of Guests (max 2 members)
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="number-pad"
          maxLength={1}
          onChangeText={[Function]}
          placeholder="Please Enter the Number of Guests"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 5,
            "marginVertical": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2c2c2c",
              "fontSize": 14,
              "paddingVertical": 5,
            }
          }
        >
          Address (max 50  )
        </Text>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Please Enter the Address"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#dadada",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
              },
              Object {
                "height": 50,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Add
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
